api/v1alpha1/envoyproxy_metric_types.go

```diff
@@ -79,4 +79,7 @@ type ProxyOpenTelemetrySink struct {
 type ProxyPrometheusProvider struct {
 	// Disable the Prometheus endpoint.
 	Disable bool `json:"disable,omitempty"`
+	// Configure the compression on Prometheus endpoint. Compression is useful in situations when bandwidth is scarce and large payloads can be effectively compressed at the expense of higher CPU load.
+	// +optional
+	Compression *Compression `json:"compression,omitempty"`
```

Comment:
the BTP API instantiated this as a list https://github.com/envoyproxy/gateway/blob/d162184e550b003c4095b49297aa6d5bee206783/api/v1alpha1/backendtrafficpolicy_types.go#L99 to support multiple compression settings, to support heterogenous clients (and compression settings), do we also need to make it a list here ?

Reply:
could 1 envoy route support multiple compression libraries? AFAIK we need to select only 1 library right?

Reply:
make it to list but limit maxitem to 1?

Reply:
that doesn't make sense IMO

Reply:
I think envoy can support multiple compressor libraries per route, which is the argument for making it a list
but realistically the persona who creates the EnvoyProxy resource with this setting, also knows beforehand the compression supported by the peer, so a single scalar value is fine here